lib/vfscore/README.md

```diff
@@ -0,0 +1,108 @@
+The `vnops` structure can be seen as the `file_operation` structure in the Linux Kernel (more as an idea).
+More about this structure [here](https://tldp.org/LDP/lkmpg/2.4/html/c577.htm).
+
+The filesystem library will define two such structures through which it will provide the specified operations.
+To understand how these operations end up being used let's examine the [`open` system call](https://github.com/unikraft/unikraft/blob/staging/lib/vfscore/syscalls.c#L111)
+
+```c
+int
+sys_open(char *path, int flags, mode_t mode, struct vfscore_file **fpp)
+{
+        struct vfscore_file *fp;
+        struct vnode *vp;
+        ...
+        error = VOP_OPEN(vp, fp);
+}
+```
+
+[`VOP_OPEN()`](https://github.com/unikraft/unikraft/blob/staging/lib/vfscore/include/vfscore/vnode.h#L219) is a macro that is defined as follows:
+
+```c
+#define VOP_OPEN(VP, FP)           ((VP)->v_op->vop_open)(FP)
+```
+
+## Register a new filesystem
+
+To map the desired function for your filesystem with the `vfscore` application, you need to have declared a `struct vnops` [structure](https://github.com/unikraft/unikraft/blob/staging/lib/ramfs/ramfs_vnops.c#L651) and populate its members, as shown below:
+
+```c
+struct vnops ramfs_vnops = {
+    ramfs_open,             /* open */
+    ramfs_close,            /* close */
+    ramfs_read,             /* read */
+    [...]
+};
+```
+
+If some of the functions are not used or not yet defined, you can use the `vfscore_vop_nullop` and cast it to the appropriate type:
+
+```c
+#define ramfs_open      ((vnop_open_t)vfscore_vop_nullop)
+#define ramfs_close     ((vnop_close_t)vfscore_vop_nullop)
+#define ramfs_seek      ((vnop_seek_t)vfscore_vop_nullop)
+```
+
+Let's see now how to link the "file operations" of a filesystem to the `vfscore` library.
+For this, the library exposes a specific structure named `vfscore_fs_type`:
+
+```c
+struct vfscore_fs_type {
+        /* name of file system */
+        const char *vs_name;
+        /* initialize routine */
+        int (*vs_init)(void);
+        /* pointer to vfs operation */
+        struct vfsops *vs_op;
+};
+```
+
+Notice that this structure contains a pointer to the `vfsops` structure, which in turn contains the `vnops` structure.
+To register a filesystem, the `vfscore` library uses an additional section in the ELF.
+You can inspect the [`extra.ld`](https://github.com/unikraft/unikraft/blob/staging/lib/vfscore/extra.ld) file to see it.
+
+As we mentioned before, these sections come with helper macros, so this time is no exception either.
+The macro that registers a filesystem is:
+
+```c
+ UK_FS_REGISTER(fssw)
+```
+
+Where the `fssw` argument is a `vfscore_fs_type` structure.
+
+Notice that the system call will eventually call the registered operation.
+So, when adding support for a new filesystem in Unikraft, you only need to implement a subset (or all) functions in the `vnops` structure.
+
```

Comment:
I would reference here the functions used by [`9pfs`](https://github.com/unikraft/unikraft/tree/staging/lib/9pfs), to provide another example (apart from [`ramfs`](https://github.com/unikraft/unikraft/tree/staging/lib/ramfs)) of implementing a filesystem.